test/common/http/conn_manager_impl_test.cc

```diff
@@ -2131,6 +2131,63 @@ TEST_F(HttpConnectionManagerImplTest, TestAccessLogWithInvalidRequest) {
   conn_manager_->onData(fake_input, false);
 }
 
+class StreamErrorOnInvalidHttpMessageTest : public HttpConnectionManagerImplTest {
+public:
+  void sendInvalidRequestAndVerifyConnectionState(bool codec_stream_error,
```

Comment:
Since these are always opposite I think you can just have one boolean, stream_error_in_invalid_messaging and make sure that the connection is terminated if !stream_error